test/c/common.c

```diff
@@ -0,0 +1,44 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+#include "common.h"
+#include <stdio.h>
+
+#ifdef _MSC_VER
+QkComplex64 make_complex_double(double real, double imag) { return (QkComplex64){real, imag}; }
+#else
+QkComplex64 make_complex_double(double real, double imag) { return real + I * imag; }
+#endif
+
+// A function to run a test function of a given name. This function will also
+// post-process the returned `TestResult` to product a minimal info message for
+// the developer running the test suite.
+int run(const char *name, int (*test_function)(void)) {
+    // TODO: we could consider to change the return value of our test functions
+    // to be a struct containing the integer return value and a custom error
+    // message which could then be included below.
+    int result = test_function();
+    int did_fail = 1;
+    char *msg;
+    if (result == Ok) {
+        did_fail = 0;
+        msg = "Ok";
+    } else if (result == EqualityError) {
+        msg = "FAILED with an EqualityError";
+    } else {
+        msg = "FAILED with unknown error";
+    }
+    fprintf(stderr, "--- %-30s: %s\n", name, msg);
```

Comment:
Would it make sense to increase the padding to a little higher than 30 since some of the methods have longer names?

Reply:
It doesn't bother me too much that the Ok's are not aligned for the longer function names. Like: https://github.com/Qiskit/qiskit/actions/runs/14495258916/job/40661239698#step:4:370 looks fine to me

We can expand it, but I'd rather do that in a separate PR. I didn't really edit the code in this function I only split it out to support multiple test files.

Reply:
That's okay then, it doesn't bother me too much either, but I wanted to bring it up here :)